pkg/resource/events-logger-secret/grafana-agent-secret.go

```diff
@@ -58,19 +45,11 @@ func GenerateGrafanaAgentSecret(lc loggedcluster.Interface, credentialsSecret *v
 
 	v, err := yaml.Marshal(values)
 	if err != nil {
-		return v1.Secret{}, errors.WithStack(err)
+		return nil, errors.WithStack(err)
 	}
 
-	secret := v1.Secret{
-		ObjectMeta: SecretMeta(lc),
-		Data: map[string][]byte{
-			"values": []byte(v),
-		},
-	}
-
-	return secret, nil
-}
+	data := make(map[string][]byte)
+	data["values"] = []byte(v)
 
-func getGrafanaAgentSecretName(lc loggedcluster.Interface) string {
-	return fmt.Sprintf("%s-%s", lc.GetClusterName(), common.GrafanaAgentExtraSecretName())
```

Comment:
Grafana agent secret uses this name https://github.com/giantswarm/observability-bundle/blob/7cfcfae127574799fa4fadc959dfe4785d32aff4/helm/observability-bundle/values.yaml#L211 you cannot remove this :D